pkg/securitycontextconstraints/sccmatching/matcher.go

```diff
@@ -160,10 +161,37 @@ func constraintSupportsGroup(group string, constraintGroups []string) bool {
 }
 
 // getNamespaceByName retrieves a namespace only if ns is nil.
-func getNamespaceByName(name string, ns *corev1.Namespace, client kubernetes.Interface) (*corev1.Namespace, error) {
+func getNamespaceByName(name string, ns *corev1.Namespace, client kubernetes.Interface, resolveUIDRange, resolveSELinuxLevel bool) (*corev1.Namespace, error) {
 	if ns != nil && name == ns.Name {
 		return ns, nil
 	}
+
+	var lastErr error
+	err := wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+		ns, lastErr = client.CoreV1().Namespaces().Get(context.TODO(), name, metav1.GetOptions{})
+		if lastErr != nil {
+			return false, nil
+		}
+
+		if _, ok := ns.GetAnnotations()[securityv1.UIDRangeAnnotation]; !ok && resolveUIDRange {
+			lastErr = fmt.Errorf("unable to find annotation %s", securityv1.UIDRangeAnnotation)
+			return false, nil
+		}
+
+		if _, ok := ns.GetAnnotations()[securityv1.MCSAnnotation]; !ok && resolveSELinuxLevel {
+			lastErr = fmt.Errorf("unable to find annotation %s", securityv1.MCSAnnotation)
+			return false, nil
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		if lastErr != nil {
+			return nil, lastErr
+		}
+		return nil, err
```

Comment:
this will return a rather generic timeout error, you may want to customize it a bit

Reply:
fixed